addon_imps/citations/mendeley.py

```diff
@@ -0,0 +1,89 @@
+import asyncio
+
+from addon_toolkit.interfaces.citation import (
+    CitationAddonImp,
+    ItemResult,
+    ItemSampleResult,
+    ItemType,
+)
+
+
+class MendeleyCitationImp(CitationAddonImp):
+    async def get_external_account_id(self, auth_result_extras: dict[str, str]) -> str:
+        async with self.network.GET(
+            "profiles/me",
+        ) as response:
+            profile_info = await response.json_content()
+            user_id = profile_info.get("id")
+            if not user_id:
+                raise KeyError("Failed to fetch user ID from Mendeley.")
+            return str(user_id)
+
+    async def list_root_collections(self) -> ItemSampleResult:
```

Comment:
I think we may have forgotten about paginations.

Reply:
as I remember we agreed to not to apply paginations in citations for now, but I will ask about that to be sure